[ARM64/Unix] LLDB

sdmaclea opened the pull request:
Patches to deal with the unfortunate fact that Arm64 LLDB is not readily available.

For ubuntu, LLDB3.9 is available in 16.10.  It is not present on earlier platforms

@jkotas

mikem8361 commented:
LGTM.

sdmaclea commented:
Do I need to mention someone to have this merged?

sdmaclea commented:
@janvorli Since you just merged a related patch...

gkhanna79 commented on cross/build-rootfs.sh:
```diff
@@ -16,7 +16,7 @@ __InitialDir=$PWD
 __UbuntuArch=armhf
 __UbuntuRepo="http://ports.ubuntu.com/"
-__LLDB_Package="lldb-3.6-dev"
+__LLDB_Package=""
```
I believe this change broke our Arm32 Ubuntu build:

```
2017-02-16T00:36:40.4286270Z -- Looking for ucol_setMaxVariable - not found
2017-02-16T00:36:40.4374030Z CMake Error at src/ToolBox/SOS/lldbplugin/CMakeLists.txt:60 (message):
2017-02-16T00:36:40.4388130Z   Cannot find lldb-3.5, lldb-3.6, lldb-3.8 or lldb-3.9.  Try installing
2017-02-16T00:36:40.4406950Z   lldb-3.6-dev (or the appropriate package for your platform)
2017-02-16T00:36:40.4414420Z 
2017-02-16T00:36:40.4420130Z 
2017-02-16T00:36:40.4432910Z -- Configuring incomplete, errors occurred!
2017-02-16T00:36:40.4445300Z See also "/root/coreclr/bin/obj/Linux.arm.Checked/CMakeFiles/CMakeOutput.log".
2017-02-16T00:36:40.4456490Z See also "/root/coreclr/bin/obj/Linux.arm.Checked/CMakeFiles/CMakeError.log".
2017-02-16T00:36:40.4468580Z ~/coreclr
2017-02-16T00:36:40.4479530Z Failed to generate CoreCLR component build project!
```

If we want to nullify lddb package for arm64, it should be done conditionally and not absolutely.

gkhanna79 replied:
@sdmaclea Can you please look into addressing this at the earliest please?

sdmaclea replied:
See #9625

gkhanna79 commented:
I think the should keep the original variable name and not add "Default" suffix since original variable name is used for pulling down packages.



Thanks,
Gaurav


-------- Original message --------
From: Steve MacLean <notifications@github.com>
Date: 2/15/17 7:48 PM (GMT-08:00)
To: dotnet/coreclr <coreclr@noreply.github.com>
Cc: "Gaurav Khanna (CLR)" <Gaurav.Khanna@microsoft.com>, Comment <comment@noreply.github.com>
Subject: Re: [dotnet/coreclr] [ARM64/Unix] LLDB (#9504)


@sdmaclea commented on this pull request.

________________________________

In cross/build-rootfs.sh<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fdotnet%2Fcoreclr%2Fpull%2F9504%23discussion_r101440712&data=02%7C01%7CGaurav.Khanna%40microsoft.com%7Ccb0ab4089df448efcfad08d4561eba18%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636228137347253970&sdata=Pur6oJFklBIZddcZ6eb%2BtpLSj2VnlKWjZ94vB9ngKmM%3D&reserved=0>:

> @@ -16,7 +16,7 @@ __InitialDir=$PWD
 __BuildArch=arm
 __UbuntuArch=armhf
 __UbuntuRepo="http://ports.ubuntu.com/"
-__LLDB_Package="lldb-3.6-dev"
+__LLDB_Package=""


See #9625<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fdotnet%2Fcoreclr%2Fpull%2F9625&data=02%7C01%7CGaurav.Khanna%40microsoft.com%7Ccb0ab4089df448efcfad08d4561eba18%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636228137347253970&sdata=opiSnlUgoNq2Oqvq5etn04cWnQTXZAbdCrq30%2F4izb0%3D&reserved=0>

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fdotnet%2Fcoreclr%2Fpull%2F9504%23discussion_r101440712&data=02%7C01%7CGaurav.Khanna%40microsoft.com%7Ccb0ab4089df448efcfad08d4561eba18%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636228137347253970&sdata=Pur6oJFklBIZddcZ6eb%2BtpLSj2VnlKWjZ94vB9ngKmM%3D&reserved=0>, or mute the thread<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAJA3wTIP8hrARb-Q5Z8heTfqQmx_UbI7ks5rc8clgaJpZM4L981S&data=02%7C01%7CGaurav.Khanna%40microsoft.com%7Ccb0ab4089df448efcfad08d4561eba18%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636228137347253970&sdata=xiMcQnuCuN7%2F0%2FN0wtJ%2FXZWEaDPCcPnMFOkH0qUHcT4%3D&reserved=0>.

sdmaclea commented:
@gkhanna79 The change that "broke" arm32 CI builds removed the **default** value for the lldb package.  Making it required for the script consumer to explicitly specify the lldb version for all platforms.  I reverted the original change and added a separate option to supress the default lldb inclusion.

sdmaclea commented:
@gkhanna79 You are right.  That was a typo....
